pkg/client/client.go

```diff
@@ -0,0 +1,128 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/go-errors/errors"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type Doer interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type Options struct {
+	BaseUrl string
+	Client  Doer
+	ApiKey  string
+}
+
+var defaultOptions = Options{
+	BaseUrl: "https://nodes.wavesnodes.com",
+	Client:  &http.Client{Timeout: 3 * time.Second},
+}
+
+type Client struct {
+	options   Options
+	Addresses *Addresses
+}
+
+type Response struct {
+	*http.Response
+}
+
+type HttpClient interface {
+}
+
+// Creates new client instance
+// If no options provided will use default
+func NewClient(options ...Options) *Client {
+
+	if len(options) > 1 {
+		panic("too many options provided. Expects no or just one item")
```

Comment:
please, don't use panics especially in places like a client library

```
panics are always fatal to your program. In panicing you never assume that your caller can solve the problem. Hence panic is only used in exceptional circumstances, ones where it is not possible for your code, or anyone integrating your code to continue.
```
[Dave Chaney](https://dave.cheney.net/2012/01/18/why-go-gets-exceptions-right)